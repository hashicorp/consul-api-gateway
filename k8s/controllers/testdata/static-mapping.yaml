apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: test-gateway
  annotations:
    "polar.hashicorp.com/node-selector": >
      {"ingress-ready":"true"}
    "polar.hashicorp.com/use-host-ports": "true"
    "polar.hashicorp.com/consul-http-address": "host.docker.internal:443"
    "polar.hashicorp.com/image": "polar:1"
    "polar.hashicorp.com/auth-method": "polar"
    "polar.hashicorp.com/service-account": "polar"
    "polar.hashicorp.com/envoy": "polar:1"
spec:
  gatewayClassName: polar
  listeners:
  - protocol: HTTP
    port: 8083
    name: http
    allowedRoutes:
      namespaces:
        from: Same
  - protocol: HTTPS
    port: 8443
    name: https
    allowedRoutes:
      namespaces:
        from: Same