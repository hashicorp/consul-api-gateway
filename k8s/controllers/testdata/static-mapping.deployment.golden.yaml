metadata:
  creationTimestamp: null
  labels:
    managedBy: polar
    name: polar-test-gateway
  name: test-gateway
spec:
  selector:
    matchLabels:
      managedBy: polar
      name: polar-test-gateway
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        managedBy: polar
        name: polar-test-gateway
    spec:
      containers:
      - command:
        - /bootstrap/polar
        - exec
        - -log-json
        - -log-level
        - info
        - -gateway-host
        - $(IP)
        - -gateway-ports
        - HTTP:http:8083,HTTPS:https:8443
        - -gateway-name
        - test-gateway
        - -consul-http-address
        - host.docker.internal:443
        - -consul-http-port
        - "8500"
        - -consul-xds-port
        - "8502"
        - -envoy-bootstrap-path
        - /bootstrap/envoy.json
        - -envoy-sds-address
        - polar-controller.default.svc.cluster.local
        - -envoy-sds-port
        - "9090"
        - -acl-auth-method
        - polar
        - -consul-ca-cert-file
        - /ca/tls.crt
        env:
        - name: IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        image: polar:1
        name: polar
        ports:
        - containerPort: 8083
          hostPort: 8083
          name: http
          protocol: TCP
        - containerPort: 8443
          hostPort: 8443
          name: https
          protocol: TCP
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      initContainers:
      - command:
        - cp
        - /bin/polar
        - /bootstrap/polar
        image: polar:1
        name: polar-init
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      nodeSelector:
        ingress-ready: "true"
      serviceAccountName: polar
      volumes:
      - emptyDir: {}
        name: bootstrap
      - emptyDir: {}
        name: certs
      - name: ca
        secret:
          secretName: consul-ca-cert
status: {}
